v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -163,7 +163,19 @@ func fuzzySetInboundNatRule(InboundNatRule genruntime.ARMResourceSpec, embeddedI
 	if err != nil {
 		return errors.Wrap(err, "unable to check that embedded inboundNatRule is the same as inboundNatRule")
 	}
-	if string(embeddedInboundNatRuleJSON) != string(inboundNatRuleJSON) {
+
+	var embeddedInboundNatRuleJSONMap map[string]interface{}
+	err = json.Unmarshal(embeddedInboundNatRuleJSON, &embeddedInboundNatRuleJSONMap)
+	if err != nil {
+		return errors.Wrap(err, "unable to check that embedded subnet is the same as subnet")
```

Comment:
Suggest using a slightly different error message for each case so that future troubleshooters know which one failed. (And other cases too.)

Reply:
Updated!